wezterm-gui/src/overlay/selector.rs

```diff
@@ -222,6 +254,9 @@ impl SelectorState {
                     key: KeyCode::Backspace,
                     ..
                 }) => {
+                    if !self.filtering && !self.selection.is_empty() {
+                        self.selection.pop();
+                    }
```

Comment:
there should probably be an `else` here, otherwise backspace will unconditionally mutate the filter term even it deleted an element from the selection

Reply:
I have added an else.